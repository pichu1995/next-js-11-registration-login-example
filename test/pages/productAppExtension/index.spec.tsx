import { useRouter } from "next/router";
import ProductAppExtension from "@pages/productAppExtension/[productId]"; 
import { render, screen } from '@test/utils';

jest.mock('@lib/hooks', () => require('@mocks/hooks'));
jest.mock('next/router', () => ({
    useRouter: jest.fn(),
}));

describe('ProductAppExtension', () => {
    const router = { query: { pid: '1' } };
    useRouter.mockReturnValue(router);

    test('renders correctly', () => {
        const { container } = render(<ProductAppExtension />);
        const panelOneHeader = screen.getByText("Basic Information");

        expect(container.firstChild).toMatchSnapshot();
        expect(panelOneHeader).toBeInTheDocument();
    });
});
