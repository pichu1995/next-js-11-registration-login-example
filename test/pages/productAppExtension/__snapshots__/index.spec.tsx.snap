// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductAppExtension renders correctly 1`] = `
<div
  class="styled__StyledBox-sc-sj5f1m-0 iiczML styled__StyledPanel-sc-1h6ef3q-0 gyZFEm"
>
  <div
    class="styled__StyledBox-sc-sj5f1m-0 gHBxjL styled__StyledFlex-sc-hcvk8l-0 hPcIGA"
  >
    <h2
      class="styled__StyledH2-sc-tqnj75-2 styled__StyledH2-sc-1h6ef3q-1 kEGvNd fjdBDH"
    >
      Basic Information
    </h2>
  </div>
  <h4
    class="styled__StyledH4-sc-tqnj75-4 XRfzZ"
  >
    Product name
  </h4>
  <p
    class="styled__StyledText-sc-tqnj75-5 dCOwSk"
  >
    Product 1
  </p>
  <h4
    class="styled__StyledH4-sc-tqnj75-4 XRfzZ"
  >
    Product type
  </h4>
  <p
    class="styled__StyledText-sc-tqnj75-5 dCOwSk"
  >
    Physical
  </p>
  <h4
    class="styled__StyledH4-sc-tqnj75-4 XRfzZ"
  >
    Default price (excluding tax)
  </h4>
  <p
    class="styled__StyledText-sc-tqnj75-5 dCOwSk"
  >
    $
    20
  </p>
  <h4
    class="styled__StyledH4-sc-tqnj75-4 XRfzZ"
  >
    Visible on storefront
  </h4>
  <p
    class="styled__StyledText-sc-tqnj75-5 dCOwSk"
  />
</div>
`;
